### PR TITLE
remove the condition DEAL_II_VERSION_GTE

### PR DESCRIPTION
**Pull request by chameerasilva**

### CODE REVIEW

File: source/particle/world.cc [hunk 1]
```diff
@@ -54,30 +54,7 @@ namespace aspect
     {
       CitationInfo::add("particles");
       if (particle_load_balancing & ParticleLoadBalancing::repartition)
-#if DEAL_II_VERSION_GTE(9,4,0)
-        this->get_triangulation().signals.weight.connect(
-#else
-        this->get_triangulation().signals.cell_weight.connect(
-#endif
```
Comment (bangerth) [inline source/particle/world.cc:61]: This isn't right. The idea for the change is that ASPECT now requires deal.II 9.4 or later, and so the `#else` branch can't happen any more. But you need to keep the `#if` branch, along with all of the code you removed from below.

File: source/particle/world.cc [hunk 2]
```diff
@@ -194,45 +171,25 @@ namespace aspect
     {
       signals.pre_refinement_store_user_data.connect(
         [&] (typename parallel::distributed::Triangulation<dim> &)
-      {
-#if DEAL_II_VERSION_GTE(9,4,0)
-        particle_handler_.prepare_for_coarsening_and_refinement();
-#else
-        particle_handler_.register_store_callback_function();
-#endif
-      });
```
Comment (bangerth) [inline source/particle/world.cc:203]: Same here and for the other changes.

**Review by bangerth — CHANGES_REQUESTED** [review]

File: source/particle/world.cc [hunk 3]
```diff
@@ -54,10 +54,9 @@ namespace aspect
     {
       CitationInfo::add("particles");
       if (particle_load_balancing & ParticleLoadBalancing::repartition)
-#if DEAL_II_VERSION_GTE(9,4,0)
+
         this->get_triangulation().signals.weight.connect(
-#else
-        this->get_triangulation().signals.cell_weight.connect(
+
 #endif
```
Comment (bangerth) [inline source/particle/world.cc:60]: If you remove the `#if` and the `#else`, then you also need to remove the `#endif`.

Same below in all of the other parts of your patch.

**Review by bangerth — CHANGES_REQUESTED** [review]

File: source/particle/world.cc [hunk 4]
```diff
@@ -54,10 +54,9 @@ namespace aspect
     {
       CitationInfo::add("particles");
       if (particle_load_balancing & ParticleLoadBalancing::repartition)
-#if DEAL_II_VERSION_GTE(9,4,0)
+
         this->get_triangulation().signals.weight.connect(
```
Comment (bangerth) [inline source/particle/world.cc:58]: Separately, remove the empty line you have here. It looks odd if to have
```
if (particle_load_balancing & ParticleLoadBalancing::repartition)

  this->get_triangulation().signals.weight.connect(
```
with an empty line in the middle.

File: source/particle/world.cc [hunk 5]
```diff
@@ -773,13 +712,12 @@ namespace aspect
     void
     World<dim>::initialize_particles()
     {
-#if !DEAL_II_VERSION_GTE(9,4,0)
+!DEAL_II_VERSION_GTE(9,4,0)
```
Comment (bangerth) [inline source/particle/world.cc:715]: You accidentally removed the `#if` here but left the condition in place. But this is not right in any case. The condition here is "if the deal.II version is NOT >=9.4", i.e., if the deal.II version is older than the minimum supported version. You can remove the whole block between `#if` and `#endif`.

**Review by bangerth — CHANGES_REQUESTED** [review]
With the exception of the one comment, this is now conceptually correct.

But if you look at https://github.com/geodynamics/aspect/pull/5198/files , there are now a lot of empty lines in more or less random places where you removed the `#if` and `#endif` lines. It might be nice to clean that up a little bit to keep the code readable.

The empty lines are probably most easily visible on the right side of this page: https://github.com/geodynamics/aspect/pull/5198/files?diff=split&w=0 where you have all of these empty green lines.

**Comment by gassmoeller** [comment]
There are still a number of new empty lines like Wolfgang mentioned above. Can you give it one more go to remove new empty lines that you added? You can use this link to find them (the green lines on the right hand side): https://github.com/geodynamics/aspect/pull/5198/files?diff=split&w=0

**Comment by gassmoeller** [comment]
Thanks, this is ready to merge now. Let's wait for the testers.

**Review by gassmoeller — APPROVED** [review]

**Comment by gassmoeller** [comment]
Congratulations to your first merged pull request to ASPECT :tada:

**Review by bangerth — APPROVED** [review]

**Comment by bangerth** [comment]
Yay, nice! Thanks for going around on this a couple of times with me!

**Pull request merged**